### PR TITLE
Feat(cli) Screenshot command macOS

### PR DESCRIPTION
**Pull request by Legitgoons**
## TLDR

Implements Issue #4493: Add comprehensive screenshot sharing functionality to Gemini CLI with `/screenshot` slash command. Enables developers to capture and automatically analyze visual issues (UI bugs, layout problems) without describing them in words. Supports multiple capture modes (fullscreen, window, interactive selection) with automatic AI analysis by default and optional capture-only mode.

**Key features:**
- 📸 **Multiple capture modes**: fullscreen, window selection, interactive selection (default)
- 🤖 **Automatic AI analysis**: Screenshots are automatically analyzed by Gemini unless `only` flag is used
- 🍏 **macOS only**: Currently implemented for macOS only using native `screencapture` command

## Dive Deeper

This PR transforms how developers communicate visual problems to Gemini CLI. Previously, developers had to describe UI issues in words ("the button is overlapping" or "the layout is broken"), which often led to misunderstandings. Now, they can simply use `/screenshot` to show exactly what they see.

### Implementation Journey

This feature was implemented through a series of focused commits:

1. **f76abddd**: Added `captureScreenshot()` function to existing `clipboardUtils.ts`
   - Leveraged existing macOS infrastructure and file management patterns
   - Used same `.gemini-clipboard` temporary directory structure
   - Started with basic interactive screenshot capture using `screencapture -i`

2. **aca8921c**: Created the `/screenshot` slash command
   - Built minimal viable command that captures and shows success/failure messages
   - Integrated with existing slash command architecture
   - Focused on core functionality first

3. **41da84ae**: Registered command in `BuiltinCommandLoader`  
   - Simple 2-line change to make command available system-wide
   - Demonstrates the power of the existing extensible architecture

4. **bb1d4fc3**: Added automatic AI analysis capability
   - Introduced `submit_prompt` return type for automatic Gemini analysis
   - Added `only` argument for capture-only mode
   - Enhanced error messages with permission guidance

5. **2c74e683**: Expanded to multiple capture modes
   - Added fullscreen (`screencapture`), window (`screencapture -w`), interactive selection modes
   - Implemented smart argument parsing for mode combinations
   - Maintained backward compatibility while adding powerful new options

6. **Security and Reliability Improvements**: A series of focused fixes to enhance command robustness
   - **ee79b2f1**: Improved argument parsing robustness for edge cases
   - **4fae2524**: Replaced exec with execFile to prevent command injection attacks
   - **6908fe0b**: Enhanced error propagation to prevent masking of file system issues
   - **efe52629**: Cleaned up error messages to remove confusing mode references
   - **64a88b41**: Aligned command description with actual implementation terminology
   - **8081a5b4**: Removed inappropriate permission guidance from cancellation messages
   - **489e7172**: Remove area mode support from screenshot command, use interactive only

7. **071332a5**: Added explicit 'interactive' mode handling and comprehensive tests for screenshot command
   - Added explicit recognition of 'interactive' as a valid mode argument in screenshot command
   - Improved mode validation consistency with explicit interactive mode handling
   - Added comprehensive test suite for screenshot command covering all functionality
   - Tested argument parsing for all capture modes (interactive, fullscreen, window)
   - Tested 'only' flag behavior for capture-without-analysis workflow
   - Tested error handling for invalid modes and screenshot capture failures
   - Ensured proper case insensitive input handling and whitespace tolerance

8. **a1e1ed9d**: Added stable tests for captureScreenshot function excluding user interaction modes
   - Added comprehensive test suite for captureScreenshot function in clipboardUtils
   - Excluded interactive and window modes that require user interaction for CI/CD stability
   - Focused tests exclusively on fullscreen mode which runs without user input
   - Tested platform compatibility ensuring null return on non-macOS systems
   - Tested error handling for invalid directories and system failures
   - Verified proper return type validation (string path or null)
   - Ensured automated test execution without blocking on user interactions
   - **Note**: Tests intentionally omit `interactive` and `window` modes to prevent CI/CD blocking on user input

### Architecture Benefits

The implementation demonstrates excellent code reuse by extending `clipboardUtils.ts` rather than creating new files. Both clipboard and screenshot features share identical patterns:
- macOS platform detection
- Temporary file management in `.gemini-clipboard`
- Same function signature: `Promise<string | null>`
- Consistent error handling and cleanup logic

This approach achieved zero code duplication while maintaining clean separation of concerns.

### Prerequisites
- macOS system 
- Terminal with Screen Recording permission (System Preferences > Security & Privacy > Privacy > Screen Recording)

### Basic Functionality Tests

1. **Interactive Selection (Default)**
   ```bash
   $ npm start
   > /screenshot
   # Should open selection crosshairs, capture selected area, auto-analyze
   ```
- Test video: [link](https://github.com/user-attachments/assets/6c80a599-d65f-4fbf-8e6b-41ac26daea62)

2. **Fullscreen Capture**
   ```bash
   > /screenshot fullscreen
   # Should instantly capture entire screen, auto-analyze
   ```
- Test video: [link](https://github.com/user-attachments/assets/db9ac5d2-9526-4aab-a3cc-5c51662f589c)

3. **Window Selection**
   ```bash
   > /screenshot window
   # Should show window selector, capture clicked window, auto-analyze
   ```
- Test video: [link](https://github.com/user-attachments/assets/9ee3c401-6b84-434e-a3ad-324d6e1515ae)

4. **Capture-Only Mode**
   ```bash
   > /screenshot only
   > /screenshot fullscreen only
   > /screenshot window only
   # Should capture but NOT auto-analyze, show file path instead
   ```
- Test video(only): [link](https://github.com/user-attachments/assets/04526d95-da28-465d-8c59-fd5a31a90898)
- Test video(fullscreen only): [link](https://github.com/user-attachments/assets/e8dbe499-65b5-4b47-9ffd-7feb92334bec)
- Test video(window only): [link](https://github.com/user-attachments/assets/121b2df5-9d6c-4cdb-905a-201c1d354259)

### Edge Cases to Test

5. **Invalid Mode Validation**
   ```bash
   > /screenshot xyz
   # Should show: Invalid capture mode: "xyz". Valid modes: fullscreen, window, interactive.
   ```
- Test video: [link](https://github.com/user-attachments/assets/d690b969-233f-4e7f-af8f-2476f8e48b1e)

6. **Cancellation Handling**
   ```bash
   > /screenshot
   # Press ESC during selection → Should show "cancelled" message
   ```
- Test video: [link](https://github.com/user-attachments/assets/087e0900-1f6a-4651-913c-86e952cf0eaa)

7. **Permission Issues**
   - Test without Screen Recording permission → Should show helpful permission guidance
   - **If you see only background/wallpaper**: Enable Screen Recording permission for your terminal in System Preferences > Security & Privacy > Privacy

8. **File Management**
   - Check `.gemini-clipboard` directory is created
   - Verify unique timestamps in filenames (`screenshot-1692123456789.png`)
   - Confirm files have content (non-zero size)

## Testing Matrix

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅  | ❓  | ❓  |
| npx      | ❓  | ❓  | ❓  |
| Docker   | ❓  | ❓  | ❓  |
| Podman   | ❓  | -   | -   |
| Seatbelt | ❓  | -   | -   |

## Linked issues / bugs
 This PR makes progress on #4493

### CODE REVIEW

File: packages/cli/src/ui/commands/screenshotCommand.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        captureMode = 'interactive';
+      } else if (argStr.includes('only')) {
+        // Handle cases like "fullscreen only", "window only"
+        shouldAnalyze = false;
+        if (argStr.includes('fullscreen')) {
+          captureMode = 'fullscreen';
+        } else if (argStr.includes('window')) {
+          captureMode = 'window';
+        }
+      }
```
[the remark this answers is not in the document]
Comment (Legitgoons) [reply packages/cli/src/ui/commands/screenshotCommand.ts]: - **Response (Commit 4fae2524)**: 
  - Replaced `exec` with `execFile` using argument arrays to prevent command injection
  - Added graceful error handling for user cancellation scenarios
  - Maintained full functionality while eliminating security risks
- **Impact**: Resolved critical security vulnerability and improved user experience

File: packages/cli/src/ui/utils/clipboardUtils.ts [hunk 2]
```diff
@@ -114,6 +114,75 @@ export async function saveClipboardImage(
...
+    }
+
+    // Clean up empty file if it exists
+    try {
+      await fs.unlink(screenshotPath);
+    } catch {
+      // Ignore cleanup errors
+    }
+
+    return null;
```
[the remark this answers is not in the document]
Comment (Legitgoons) [reply packages/cli/src/ui/utils/clipboardUtils.ts]: - **Response (Commit ee79b2f1)**:
  - Implemented structured argument parsing with proper validation
  - Added clear error messages for invalid mode combinations
  - Enhanced robustness for edge cases and user input errors
- **Impact**: Ensured predictable command behavior and better user feedback

**Comment by bdmorgan** [comment]
Hi @Legitgoons, thank you so much for your contribution to Gemini CLI! We really appreciate the time and effort you've put into this.

We're making some updates to our contribution process to improve how we track and review changes. Please take a moment to review our recent discussion post: [Improving Our Contribution Process & Introducing New Guidelines](${DISCUSSION_URL}).

Key Update: Starting **January 26, 2026**, the Gemini CLI project will require all pull requests to be associated with an existing issue. Any pull requests not linked to an issue by that date will be automatically closed.

Thank you for your understanding and for being a part of our community!
